### DISCUSSION TITLE
Why does VSCode have custom EventEmitters?

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#1028] [Extension Development QnA] [answered]
This is more a background question, but why does VSCode have its own EventEmitter class? Why doesn't it just use node ones?

**Comment by tamuratak** [comment, marked as answer]
It would make it possible to run VS Code on browsers, e.g. [vscode.dev](https://code.visualstudio.com/docs/editor/vscode-web). EventEmitter isn't supported on browsers.
